makiflow/trainers/nn_render/render_trainer.py

```diff
@@ -252,20 +274,98 @@ def _restore_model(self, exp_params):
+
+        self._test_model.set_session(self._sess)
+        if weights_path is not None:
+            self._test_model.load_weights(weights_path, layer_names=pretrained_layers)
+
     # -----------------------------------------------------------------------------------------------------------------
     # -----------------------------------EXPERIMENT UTILITIES----------------------------------------------------------
 
+    def save_texture_info_to_scv(self, x, folder):
+        print('Prepare data for saving texture info into csv...')
+        std = [x[i, ...].std().astype(np.float32) for i in range(x.shape[0])]
+        mean = [x[i, ...].mean().astype(np.float32) for i in range(x.shape[0])]
+        assym = [np.array(skew(x[i, ...], axis=None)).astype(np.float32) for i in range(x.shape[0])]
+        kurt = [np.array(kurtosis(x[i, ...], fisher=True, axis=None)).astype(np.float32) for i in range(x.shape[0])]
+
+        df = pd.DataFrame({
+            'std': std,
+            'mean': mean,
+            'kurt': kurt,
+            'assym': assym,
+        })
+        df.to_csv(folder)
+
     def _perform_testing(self, exp_params, save_path, path_to_weights, FPS=25):
         # Create test video from pure model.
         # NOTICE output of model and input image size (not UV) must be equal
         print('Testing the model...')
-        print('Collecting predictions...')
 
         # load weights to test model
         self._test_model.load_weights(path_to_weights)
 
         # Collect data and predictions
 
+        self._record_video(exp_params=exp_params, save_path=save_path, FPS=FPS)
+        self._plot_values(exp_params=exp_params, save_path=save_path)
+
+    def _plot_values(self, exp_params, save_path):
+        print('Prepare to plot values...')
+
+        # Plot product of the neural network
+        path = exp_params[ExpField.PATH_TEST_UV][int(np.random.choice(len(exp_params[ExpField.PATH_TEST_UV]), 1))]
+        random_number = int(np.random.choice(len(path + '/*.npy'), 1))
+        uv = np.load(path + '/' + str(random_number) + '.npy').astype(np.float32)
+        uv = uv.reshape(1, *uv.shape)
+        origin_uv = copy.deepcopy(uv)
+
+        for _ in range(exp_params[ExpField.BATCH_SIZE] - 1):
+            uv = np.concatenate((origin_uv, uv), 0)
+
+        values = []
+
+        for name_layer in exp_params[ExpField.PLOT_VALUE_LAYERS]:
+            tensor_of_layer = self._test_model.get_node(name_layer).get_data_tensor()
+            values.append(self._sess.run(tensor_of_layer,
+                                         feed_dict={self._test_model._input_data_tensors[0]: uv})[0])
+
+        TestVisualizer.plot_numpy_dist_obs(values=values, legends=exp_params[ExpField.PLOT_VALUE_LAYERS],
+                                           save_path=save_path + 'histograms_of_layers.png',
+        )
+
+        # Plot weights of the texture
```

Comment:
The texture itself is not being plotted, therefore, the commentary is confusing. Fix the commentary.